### ISSUE TITLE
Namespace member functions links are broken (Origin: bugzilla #790856)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#6291]
status NEW severity *normal* in component *general* for *---*
Reported in version *1.8.13* on platform *Other*
Assigned to: Dimitri van Heesch

On 2017-11-26 16:39:51 +0000, lectem@gmail.com wrote:

> Functions in namespaces will not be linked to properly.
> 
> The html anchor generated contains a file_ prefix before the identifier (see https://github.com/doxygen/doxygen/blob/master/src/memberdef.cpp#L2582 ) which causes all links to these function to not work.
> 
> Example of "More..." link
> 
> _file_stream_8h.html#a11436006222ebf8024ba674143cb42ad
> 
> The actual html code of the links :
> 
> <tr class="memitem:a11436006222ebf8024ba674143cb42ad"><td class="memItemLeft" align="right" valign="top">void&# 160;</td><td class="memItemRight" valign="bottom"><a class="el" href="_file_stream_8h.html#a11436006222ebf8024ba674143cb42ad">WorldStone::foo</a> ()</td></tr>
> <tr class="memdesc:a11436006222ebf8024ba674143cb42ad"><td class="mdescLeft">&# 160;</td><td class="mdescRight">Test function.  <a href="_file_stream_8h.html#a11436006222ebf8024ba674143cb42ad">More...</a><br /></td></tr>
> 
> The html code for the function (with the anchor)
> 
> </div><h2 class="groupheader">Function Documentation</h2>
> <a id="file_a11436006222ebf8024ba674143cb42ad"></a>
> <h2 class="memtitle"><span class="permalink"><a href="#file_a11436006222ebf8024ba674143cb42ad">&# 9670;&nbsp;</a></span>foo()</h2>
> 
> 
> Can be reproduced with the following file :
> 
> 
> /** 
>  * @file FileStream.h
>  */
> 
> #pragma once
> 
> namespace YourNamespace
> {
> /** @brief Test function.
>  *
>  * Some advanced desc
>  */
> void foo();
> }
> 
> Occurs at least on 1.8.12-13 but I think it has been here for a while.

**Comment by pinchartl** [comment]
I've just hit the same issue and can thus confirm the bug. I could try to submit a fix, but I'm not sure to understand the need for the "file_" prefix. git blame report that the prefix was introduced in

commit 3aaaf142be657201a2664c10b96447acb1610187
Author: Dimitri van Heesch <dimitri@stack.nl>
Date:   Tue Aug 24 20:00:14 2004 +0000

    Release-1.3.8-20040824

but the commit isn't very helpful.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7319

**Comment by pinchartl** [comment]
Thank you. I've continued investigating this issue in parallel, and it turns out that it was at least partly related to the namespace in which the target variable resides not being documented. After documenting the namespace, links were properly generated.

While this behaviour is properly correct, it is quite confusing, as doxygen doesn't emit any warning, and generates documentation for the variables. Only links are silently broken. I thus wonder if the right fix is to add an exception for C++ as proposed in https://github.com/doxygen/doxygen/pull/7319, or to warn about the missing namespace documentation (or skip generating documentation for namespaces that are not documented).

**Comment by albert-github** [comment]
The later, warning, might be a good idea, please file an issue report for it with attached a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by pinchartl** [comment]
On Sat, Oct 12, 2019 at 10:25:29AM -0700, albert-github wrote:
> The later, warning, might be a good idea, please file an issue report for it
> with attached a, small, self contained example (source+configuration file in a
> tar or zip) that allows us to reproduce the problem? Please don't add external
> links as they might not be persistent.

I will do so. I got a bit sidetracked by a separate issue that happened
to be triggered by the exact same piece of code :-) It has already been
reported in https://github.com/doxygen/doxygen/issues/5423, so I will
strip the example down to only focus on the namespace problem.

-- 
Regards,

Laurent Pinchart
